### PR TITLE
mavlink receiver fixes

### PR DESCRIPTION
**Pull request by julianoes**
This contains various cleanup in mavlink and mavlink receiver the most notable being a different handling of how to keep track of sequence numbers for various components.

Extraced out of https://github.com/PX4/PX4-Autopilot/pull/17404.

### CODE REVIEW

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -3230,17 +3063,80 @@ MavlinkReceiver::Run()
...
-void
-MavlinkReceiver::receive_start(pthread_t *thread, Mavlink *parent)
+void MavlinkReceiver::print_detailed_rx_stats() const
+{
+	const uint32_t now_ms = hrt_absolute_time() / 1000;
+
+	// TODO: add mutex around shared data.
+	for (unsigned i = 0; i < MAX_REMOTE_COMPONENTS; ++i) {
+		if (_component_states[i].received_messages > 0) {
+			printf("\t  received from %u/%u: %lu, lost: %lu, last %lu ms ago\n",
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp:3126]:
```suggestion
			printf("\t  received from SYSID:%u COMPID:%u: %lu, lost: %lu, last %lu ms ago\n",
```
Add something to show what these numbers are?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -3230,17 +3063,80 @@ MavlinkReceiver::Run()
...
+	}
 }
 
-void
-MavlinkReceiver::receive_start(pthread_t *thread, Mavlink *parent)
+void MavlinkReceiver::print_detailed_rx_stats() const
+{
+	const uint32_t now_ms = hrt_absolute_time() / 1000;
+
+	// TODO: add mutex around shared data.
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp:3123]: You could use `hrt_elapsed_time_atomic` for the last received and pretty much ignore the rest that are already small enough to be atomic.

Comment (dagar) [reply src/modules/mavlink/mavlink_receiver.cpp:3123]: Nevermind, just saw that you're storing the time in a uint32 as milliseconds.

Comment (julianoes) [reply src/modules/mavlink/mavlink_receiver.cpp:3123]: So you're saying the copies below are not a problem ever?

Comment (dagar) [reply src/modules/mavlink/mavlink_receiver.cpp:3158]: Yes, for the sake of printing I think it's acceptable and not worth additional locking.

**Review by dagar — APPROVED** [review]

**Pull request merged**